deps/streamsearch/sbmh.js

```diff
@@ -196,14 +196,15 @@ SBMH.prototype._sbmh_feed = function (data) {
   ) {
     ++pos
   }
-  if (pos < len) {
-    data.copy(this._lookbehind, 0, pos, pos + (len - pos))
+
+  // Copy the trailing haystack data that we cannot match yet to the lookbehind buffer.
+  if (pos !== len) {
+    data.copy(this._lookbehind, 0, pos, len)
     this._lookbehind_size = len - pos
   }
 
   // Everything until pos is guaranteed not to contain needle data.
-  if (pos > 0) { this.emit('info', false, data, this._bufpos, pos < len ? pos : len) }
-
+  if (pos !== 0) { this.emit('info', false, data, this._bufpos, pos) }
```

Comment:
Here, pos can never surpass len, because it starts smaller than len, and it's incremented by 1 until it reaches len or a matching subpart is found